syntax="proto3";
package eos.console;

message FsProto {
  message AddProto {
    bool Manual = 1;
    uint64 Fsid  = 2;
    uint64 Port = 3;
    string Uuid = 4;
    string NodeQueue = 5;
    string HostPort = 6;
    string Mountpoint = 7;
    string Schedgroup = 8;
    string Status = 9;
  }

  message BootProto {
    bool SyncMgm = 1;
    oneof id {
      uint64 Fsid = 2;
      string NodeQueue = 3;
    }
  }

  message ConfigProto {
    string Key = 1;
    string Value = 2;
    oneof id {
      uint64 Fsid = 3;
      string Uuid = 4;
      string HostPortPath = 5;
    }
  }

  message DropDeletionProto {
    uint64 Fsid = 1;
  }

  message DropFilesProto {
    int64 Fsid = 1;
    bool Force = 2;
  }

  message DumpMdProto {
    enum DisplayMode {
      DEFAULT = 0;
      SILENT  = 1;
      MONITOR = 2;
    }

    uint64 Fsid = 1;
    DisplayMode display = 2;
    bool ShowFid = 3;
    bool ShowPath = 4;
  }

  message MvProto {
    string Src = 1;
    string Dst = 2;
  }

  message LsProto {
    enum DisplayMode {
      DEFAULT = 0;
      MONITOR = 1;
      LONG    = 2;
      ERROR   = 3;
      IO      = 4;
      FSCK    = 5;
      DRAIN   = 6;
   }

   DisplayMode display = 1;
   bool Silent = 2;
   bool Brief = 3;
   string Matchlist = 4;
  }

  message RmProto {
    oneof id {
      uint64 Fsid = 1;
      string NodeQueue = 2;
    }
  }

  message StatusProto {
    oneof id {
      uint64 Fsid = 1;
      string HostMountpoint = 2;
    }

    bool LongFormat = 3;
    bool RiskAssesment = 4;
  }

  oneof subcmd {
    AddProto add = 1;
    BootProto boot = 2;
    ConfigProto config = 3;
    DropDeletionProto dropdel = 4;
    DropFilesProto dropfiles = 5;
    DumpMdProto dumpmd = 6;
    MvProto mv = 7;
    LsProto ls = 8;
    RmProto rm = 9;
    StatusProto status = 10;
  }
}
